### PR TITLE
documented sample for serial service

### PR DESCRIPTION
**Pull request by NeoCortex97**
I wrote documentation for my sample that Is available in pr [#132](https://github.com/codeoverflow-org/nodecg-io/pull/132) in nocdecg-io

### CODE REVIEW

File: docs/samples/serial.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+## Using the Twitch sample bundle
```
Comment (daniel0611) [inline docs/samples/serial.md:1]: Twitch copy-pastes are a meme at this point :laughing: 

```suggestion
## Using the serial sample bundle
```

File: docs/samples/serial.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+The serial example bundle in `samples/serial` demonstrates the ability to exchange data with a device that is connected via serial. Here is a guide to how to get it working.
+
+### Prerequisites
+
+-   Working NodeCG & nodecg-io installation
+-   An Arduino or any other microcontroller development board that can send and receive data via serial.
+
+_Note:_ If you don't have such a key yet you can generate it on https://twitchapps.com/tmi/. Just log into your Twitch account and copy the token.
+
```
Comment (daniel0611) [inline docs/samples/serial.md:11]: More twitch stuff
```suggestion
```

File: docs/samples/serial.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+### Configure the Twitch sample bundle
+
+1. A load a simple serial echo sketch on your microcontroller. You can find a working arduino sketch at the end of this article.
```
Comment (daniel0611) [inline docs/samples/serial.md:14]:
```suggestion
1. Load a simple serial echo sketch on your microcontroller. You can find a working arduino sketch at the end of this article.
```

File: docs/samples/serial.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+3. Go to the `nodecg-io` tab in the nodecg dashboard.
+
+4. Login using your password. If this is your first run, then enter the password with which you want to encrypt your configurations and credentials.
+
+5. Create a new serial service instance using the left upper menu.
+
+6. Enter the information of your device.
+
+    The created instance should be automatically selected, if not select it in the upper left menu. Enter the com port or other identifying information in one of these format:
```
Comment (daniel0611) [inline docs/samples/serial.md:26]:
```suggestion
    The created instance should be automatically selected, if not select it in the upper left menu. Enter the com port or other identifying information in one of these formats:
```

File: docs/samples/serial.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            inputString += inChar;
+            if (inChar == '\n') {
+            stringComplete = true;
+        }
+    }
+    if (stringComplete) {
+        Serial.println("pong");
+        stringComplete = false;
+    }
+  delay(1);        // delay in between reads for stability
```
Comment (daniel0611) [inline docs/samples/serial.md:81]: Align all lines to have 4x space indentation
```suggestion
    delay(1);        // delay in between reads for stability
```

File: mkdocs.yml [hunk 6]
```diff
@@ -57,7 +57,8 @@ nav:
           - Rcon sample: samples/rcon.md
           - sACN:
                 - sACN Receiver sample: samples/sacn-receiver.md
-                - sACN Sender sample: samples/sacn-sender.md
+                - sACN Receiver sample: samples/sacn-sender.md
+          - Serial anmple: samples/serial.md
```
Comment (daniel0611) [inline mkdocs.yml:61]:
```suggestion
          - Serial sample: samples/serial.md
```

File: mkdocs.yml [hunk 7]
```diff
@@ -57,7 +57,8 @@ nav:
           - Rcon sample: samples/rcon.md
           - sACN:
                 - sACN Receiver sample: samples/sacn-receiver.md
-                - sACN Sender sample: samples/sacn-sender.md
+                - sACN Receiver sample: samples/sacn-sender.md
```
Comment (daniel0611) [inline mkdocs.yml:60]: sACN Sender was correct here. The receiver is one line above and also both filenames are correct.
```suggestion
                - sACN Sender sample: samples/sacn-sender.md
```

**Review by daniel0611 — CHANGES_REQUESTED** [review]

File: docs/samples/serial.md (hunk 1 above)
Comment (NeoCortex97) [reply docs/samples/serial.md:1]: Thanks, should have grepped before opening the pr

File: docs/samples/serial.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    _Note:_ If you want to list all connected devices you can open a terminal in `nodecg-io-serial` and run `npm run list`.
+    _Note:_ If you are using multiple devices you might want to use the pnpId, since ports can change between reboots!
+
+7. Set the created serial service instance to the service dependency of the serial bundle.
+
+    Select the serial bundle and the serial service in the left bottom menu and then select the service instance that should be used by the serial bundle (in this case the name of the previously created serial instance).
+
+8. Check the nodecg logs
+
+    You should see an error or a success message and nodecg-io will send ping to the microcontroller every 10 seconds. The adruino device will respond with pong. you ahould see the pong message displayed in the log.
```
Comment (daniel0611) [inline docs/samples/serial.md:54]:
```suggestion
    You should see an error or a success message and nodecg-io will send ping to the microcontroller every 10 seconds. The arduino device will respond with pong. You should see the pong message displayed in the log.
```

File: docs/samples/serial.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+    If
```
Comment (daniel0611) [inline docs/samples/serial.md:56]: I think you meant to write something after this 😉

**Review by daniel0611 — CHANGES_REQUESTED** [review]

File: docs/samples/serial.md (hunk 9 above)
Comment (NeoCortex97) [reply docs/samples/serial.md:56]: yup, ill fix it.

**Review by daniel0611 — APPROVED** [review]

**Pull request merged**